### PR TITLE
[ipmitool] Include channel, raw, and lan information.

### PR DESCRIPTION
**Pull request by vikas-goel**
Signed-off-by: Vikas Goel <vikas.goel@gmail.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

File: sos/report/plugins/ipmitool.py [hunk 1]
```diff
@@ -27,12 +27,17 @@ def setup(self):
             cmd += " -I usb"
 
         self.add_cmd_output([
+            "%s channel info 3" % cmd,
+            "%s channel getaccess 3" % cmd,
+            "%s raw 0x30 0xb0" % cmd,
+            "%s raw 0x30 0x65" % cmd,
```
Comment (TurboTurtle) [inline sos/report/plugins/ipmitool.py:33]: I'm not familiar with what `0x30`, `0xb0`, or `0x65` map to. I know they're ipmi commands, but I'm just not sure what commands they are.

Can you please detail what these are? We'd likely be well served sticking that in a comment as well.

File: sos/report/plugins/ipmitool.py [hunk 2]
```diff
@@ -27,12 +27,17 @@ def setup(self):
             cmd += " -I usb"
 
         self.add_cmd_output([
+            "%s channel info 3" % cmd,
+            "%s channel getaccess 3" % cmd,
```
Comment (TurboTurtle) [inline sos/report/plugins/ipmitool.py:31]: What's the output of this command? I was under the impression that `getaccess` had to do with authorizing access, which is not something we'd want to automatically do when a user runs `sos`.

**Comment by TurboTurtle** [comment]
Also restarted this 18.04 timeout, again I think this is the same transient timeouts we've seen in the past.

File: sos/report/plugins/ipmitool.py (hunk 2 above)
Comment (vikas-goel) [reply sos/report/plugins/ipmitool.py:31]:
```
$ cat ipmitool_channel_info_3
Channel 0x3 info:
  Channel Medium Type   : 802.3 LAN
  Channel Protocol Type : IPMB-1.0
  Session Support       : multi-session
  Active Session Count  : 13
  Protocol Vendor ID    : 7154
  Volatile(active) Settings
    Alerting            : enabled
    Per-message Auth    : enabled
    User Level Auth     : enabled
    Access Mode         : always available
  Non-Volatile Settings
    Alerting            : enabled
    Per-message Auth    : enabled
    User Level Auth     : enabled
    Access Mode         : always available
$ cat ipmitool_channel_getaccess_3
Maximum User IDs     : 15
Enabled User IDs     : 3

User ID              : 1
User Name            :
Fixed Name           : Yes
Access Available     : call-in / callback
Link Authentication  : disabled
IPMI Messaging       : disabled
Privilege Level      : NO ACCESS
Enable Status        : disabled

User ID              : 2
User Name            : root
Fixed Name           : No
Access Available     : call-in / callback
Link Authentication  : enabled
IPMI Messaging       : enabled
Privilege Level      : ADMINISTRATOR
Enable Status        : disabled
...
```

It does not store password.

File: sos/report/plugins/ipmitool.py (hunk 1 above)
Comment (vikas-goel) [reply sos/report/plugins/ipmitool.py:33]: From Intel's guide:
`0x30 0x65 = Get HDD drive Fault LED State`
This command retrieves the drive LED states for each slot. The second returned data is in the same format as the first request data in the Set HDD drive fault LED command.

`0x30 0xb0 = Get LED Status`
Retrieves the LED status.
Status definitions:00b = Off01b = Blink10b = On11b = Invalid (not installed)

**Comment by vikas-goel** [comment]
Hi @TurboTurtle ,
Added details of the commands.

**Comment by TurboTurtle** [comment]
> Hi @TurboTurtle ,
> Added details of the commands.

If you could add those as comments in the plugin, that'd be ideal so we have a reference to it beyond this PR. Once that's done this looks good to me.

**Comment by vikas-goel** [comment]
> > Hi @TurboTurtle ,
> > Added details of the commands.
> 
> If you could add those as comments in the plugin, that'd be ideal so we have a reference to it beyond this PR. Once that's done this looks good to me.

Done. The build is failing for unrelated issue.

**Review by TurboTurtle — APPROVED** [review]

**Comment by vikas-goel** [comment]
@pmoravec , please review

**Review by pmoravec — APPROVED** [review]

**Pull request merged**